### PR TITLE
Issue #9 - Sidebar added

### PR DESCRIPTION
**Pull request by naarrek**

### CODE REVIEW

File: config.js
Comment (Manvel) [inline config.js]: Suggestion: Let's rename `nav` to  something more meaningful, as we are expecting nav to be a page queried by `queryPages`, maybe we can name it just `page` from `queryPages`?

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [inline theme/layouts/_sidebar.ejs]: Suggestion: As it's a method let's call it for example `getSubPages` in order to make it clear that's not array of subPages. Like already result of calling `site.getSubPages()`

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [inline theme/layouts/_sidebar.ejs]: I assume here you wanted to create the full navigation tree instead of adding just 1 level, maybe:
```
const currentOriginalPath = "documentation/actions/check";
const navigation = [];
for (const menuItem of site.menuItems) {
  const page = query(({originalPathname}) => originalPathname === menuItem)[0];
  if (menuItem === currentOriginalPath.split("/")[0]) {
    page.sub_items = subPages(page);
  }
  navigation.push(page);
}
```

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [inline theme/layouts/_sidebar.ejs]: Not sure if this is necessary if you will general `navigation` as suggested in the above comment, instead you will only need current page's `originalPathname` in order to not generate children of current page.

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [inline theme/layouts/_sidebar.ejs]: We don't need to specify `.primary-menu` class to style top most list items to do that you can just use direct children selector `#sidebar ul > li`.

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [inline theme/layouts/_sidebar.ejs]: Not sure if this logic is required, here we need to ensure that item has subItems and those subItems are not children of current page.
```
if (sub_items && originalPathname !== currentOriginalPath) {
  recNav(sub_items);
}
```

File: theme/layouts/_sidebar.ejs
Comment (Manvel) [reply theme/layouts/_sidebar.ejs]: Or you can filter out children of current page and reuse existing `recNav` method from `_navigation.ejs`.

File: theme/layouts/_breadcrumbs.ejs
Comment (Manvel) [inline theme/layouts/_breadcrumbs.ejs]: Detail: I assume change in this file was unintentional.

**Review by Manvel — CHANGES_REQUESTED** [review]
I can see that each time you create a PR it requires less and less comments. Good job.

I haven't reviewed the CSS part, yet, but I'll do it after we finish with the Markup part.

File: theme/layouts/_sidebar.ejs
Comment (naarrek) [reply theme/layouts/_sidebar.ejs]: I think this does not replicate the Drupal website working logic. Not only `currentOriginalPath` matters for viewing children but also we need to specify the logic for "current" parent to view it's sections.

File: config.js
Comment (naarrek) [reply config.js]: Is it good idea considering that page is reserved by `page.title, page.pathname, ...` varible?

Comment (Manvel) [reply config.js]: Good point, as `page.title` is only accessible in the EJS files I think we are good to go, also if we would have another page variables accessible in the parent scope of the function using local `page` would be fine in this case as you would refer to the local variable not the one specified in the upper scope. So I think we are safe to use `page` here, but feel free to use other meaningful variable if you think of one.

Comment (Manvel) [inline config.js]: Suggestion: We can also introduce second parameter `level`. To help with sidebar implementation, something like:
```
getSubPages: function (nav, level = 100) {
  ...
  if (subPages.length && --level)
  ...
}
```

This way we should be able to get only children of the existing page, which can be useful in sidebar, ex.:
```
site.getSubPages(otherPage, 1);
```

Comment (Manvel) [reply config.js]: Or maybe:
```
getSubPages: function (nav, grandChildren) {
...
  if (subPages.length && grandChildrenl)
...
    templateData.site.getSubPages(subPage, grandChildren);
```

File: theme/less/main.less
Comment (Manvel) [inline theme/less/main.less]: Detail: Seems like we are already specifying this style outside of `@media` query, I guess it's safe to remove it.

File: theme/less/_sidebar.less
Comment (Manvel) [inline theme/less/_sidebar.less:4]: I did notice that we are using precise pixels in some places and relative em/rem in other places, thinking we might need to adjust in future, to make sizing relative which might help with specifying variables.

I have created this -> https://github.com/browser-automation/cba-website/issues/28 

I guess we will tackle that at one point, but it shouldn't be blocker for this issue I guess.

**Review by Manvel — APPROVED** [review]
Good job Narek jan. There is just 1 detail comment, which will be great to address before merging, other than that this is ready.

**Pull request merged**
